packages/typespec-ts/src/index.ts

```diff
@@ -245,6 +245,12 @@ export async function $onEmit(context: EmitContext) {
     const hasPackageFile = await existsSync(
       join(dpgContext.generationPathDetail?.metadataDir ?? "", "package.json")
     );
+    const hasTestFolder = await fsextra.pathExists(
+      join(dpgContext.generationPathDetail?.metadataDir ?? "", "test")
+    );
+    option.generateTest =
+      option.generateTest === true ||
+      (option.generateTest === undefined && !hasTestFolder && isBranded);
```

Comment:
as this relies on the folder exist logic, we will update generateTest before we generate the metadata.